### ISSUE TITLE
The API `eth_getTransactionCount(..., "pending")` is not synchronous with the API `eth_sendRawTransaction`.

### ISSUE THREAD

**Issue by chaoticlonghair** [axonweb3/axon#1544]
### Description

- Write a script with your favorite language.
  - Get the nonce `M` with pending transactions for a sender through the JSON-RPC.
  - Construct a transaction.
  - Send the transaction.
  - Get the nonce `N` with pending transactions for a sender through the JSON-RPC immediately, do NOT wait for the transaction to be mined.
  - Check if `M` is equal to `N`.

- Start a Ethereum dev chain with [Geth] and parameters `--dev --dev.period 3`.

  - Run the script.

- Start a Axon dev chain with default configuration `[params.consensus_config] interval = 3000`.

  - Run the script.

- Compare the results of above tests:

  - No failed tests on Ethereum.
  - More than 50% tests on Axon are failed.

- Base on the previous fact, on Axon, transactions could NOT be sent one by one immediately, because the API `eth_getTransactionNonce(..., "pending")` is not synchronous with the API `eth_sendRawTransaction`.

[Geth]: github.com/ethereum/go-ethereum

**Comment by chaoticlonghair** [comment]
Even we try to fix it with a tricky way as following:

https://github.com/axonweb3/axon/blob/76108903486e7e2d32a7343096e0af765a01bb45/core/api/src/jsonrpc/impl/web3.rs#L228-L230

But it isn't fixed completely, just increased the chance for success. 

I do the test as the issue described again, the result is:
- For Axon: `Total: 100, passed: 65, failed: 35.`
- For Geth: `Total: 100, passed: 100, failed: 0.`

That's why some CI checks failed randomly.

**Comment by Flouse** [comment]
> That's why some CI checks failed randomly.

- Do you mean https://github.com/axonweb3/axon/issues/1579 ?

**Comment by chaoticlonghair** [comment]
> > That's why some CI checks failed randomly.
> 
>     * Do you mean [Sometimes the test `eth_getCode › eth_getCode_9` failed by accident #1579](https://github.com/axonweb3/axon/issues/1579) ?

No, I just guess it is related to "v3_core_test" timeout or some other timeouts.

I didn't check #1579.

**Comment by Flouse** [comment]
> No, I just guess it is related to "v3_core_test" timeout or some other timeouts.

The [v3-core-test (ubuntu-22.04)](https://github.com/axonweb3/axon/actions/runs/6946677952/job/18898734858#step:15:860) action has timed out.
cc @sunchengzhu
